src/karapace/schema_registry_apis.py

```diff
@@ -446,13 +396,18 @@ async def compatibility_check(
                 status=HTTPStatus.INTERNAL_SERVER_ERROR,
             )
 
-        result = check_compatibility(
-            old_schema=old_schema,
-            new_schema=new_schema,
-            compatibility_mode=compatibility_mode,
-        )
+        new_schema = self.get_new_schema(request.json, content_type)
+        old_schema = self.get_old_schema(subject, Versioner.V(version), content_type)
+        if compatibility_mode.is_transitive():
+            # Ignore the schema version provided in the rest api call (`version`)
+            # Instead check against all previous versions (including `version` if existing)
+            result = self.schema_registry.check_schema_compatibility(new_schema, subject)
+        else:
+            # Check against the schema version provided in the rest api call (`version`)
+            result = SchemaCompatibility.check_compatibility(old_schema, new_schema, compatibility_mode)
+
         if is_incompatible(result):
-            self.r({"is_compatible": False}, content_type)
+            self.r({"is_compatible": False, "incompatibilities": f"{', '.join(result.messages)[:300]}"}, content_type)
```

Comment:
again super nit, I would declare the error message as a separate variable like:
```python3
maybe_truncated_error = ', '.join(result.messages)[:300]
self.r({"is_compatible": False, "incompatibilities": maybe_truncated_error}, content_type)
```

Reply:
Agree, looks more readable now!